src/core/solvers/ddp.cpp

```diff
@@ -172,6 +181,9 @@ double SolverDDP::calcDiff() {
     const std::size_t T = problem_->get_T();
     const std::vector<boost::shared_ptr<ActionModelAbstract> >& models = problem_->get_runningModels();
     const std::vector<boost::shared_ptr<ActionDataAbstract> >& datas = problem_->get_runningDatas();
+#ifdef CROCODDYL_WITH_MULTITHREADING
+#pragma omp parallel for num_threads(NUM_THREADS)
```

Comment:
Does `could_be_feasible` here get reduced correctly as it's shared (not declared private) across threads?

Reply:
We can remove the `if(could_be_feasible)` from inside the `for loop`, and do the check separately afterwards. The multithreaded loop would then only handle the statement `model->get_state()->diff(xs_[t + 1], d->xnext, fs_[t + 1]);
`
@cmastalli could you do these changes here?

Reply:
Done!